src/PrestaShopBundle/DependencyInjection/Compiler/LoadServicesFromModulesPass.php

```diff
@@ -83,5 +84,24 @@ public function process(ContainerBuilder $container)
                 $loader->load('services.yml');
             }
         }
+
+        //@todo: POC proposal. If accepted it probably needs to be moved to dedicated pass, but priority is important
+        //       (it should probably be after this ModulesPass. e.g. using same in ContainerInjectionPass doesn't work)
+        // we find all definitions in order to search all decorated services
+        // @todo: find a way to optimize and find all decorated services or all controllers (cannot find decorated controllers by tags)
+        $allDefinitions = $container->getDefinitions();
+        foreach ($allDefinitions as $definition) {
+            $decoratedService = $definition->getDecoratedService();
+            // if service is decorator
+            if ($decoratedService && $container->hasDefinition($decoratedService[0])) {
+                $decoratedServiceDefinition = $container->getDefinition($decoratedService[0]);
+                // and if service is controller
+                if ($decoratedServiceDefinition->hasTag('controller.service_arguments')) {
+                    // then we add controller.service_arguments tag to the decorator and remove it from the decorated class
+                    $definition->addTag('controller.service_arguments');
```

Comment:
```suggestion
                    if (!$definition->hasTag('controller.service_arguments')) {
                        $definition->addTag('controller.service_arguments');
                    }
```

Reply:
Maybe we should check that the decorator doesn't already have the tag in case it was added manually in its definition